### PR TITLE
Add failure handling to REST Call

### PR DESCRIPTION
**Pull request by kenfinnigan**

### CODE REVIEW

**Comment by kenfinnigan** [comment]
Interesting side effect I found in testing this, depending on what the REST timeout call is set to, you may still get "FAILED" even though http-barista is there because of the random delay in responding.

Think it highlights everything nicely

File: coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java [hunk 1]
```diff
@@ -24,7 +25,14 @@
 
     @PostMapping("/http")
     public Uni<Beverage> http(Order order) {
-        return barista.order(order.setOrderId(getId()));
+        return barista.order(order.setOrderId(getId()))
+                .onItem().invoke(beverage -> beverage.setPreparationState(Beverage.State.READY))
+                .ifNoItem().after(Duration.ofMillis(1500)).recoverWithItem(createFallbackBeverage(order))
```
Comment (cescoffier) [inline coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java]: let's have some fun...

```java
.onItem().invoke(beverage -> beverage.setPreparationState(Beverage.State.READY))
                .ifNoItem().after(Duration.ofMillis(1500)).fail()
                .onFailure().recoverWithItem(createFallbackBeverage(order));
```

**Review by cescoffier — CHANGES_REQUESTED** [review]

File: coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java (hunk 1 above)
Comment (kenfinnigan) [reply coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java]: does `.fail()` then trigger the `onFailure()`?

Comment (cescoffier) [reply coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java]: Yes, that the beauty of it :-)

Comment (kenfinnigan) [reply coffeeshop-service/src/main/java/me/escoffier/quarkus/coffeeshop/CoffeeShopResource.java]: That's pretty sweet

**Review by cescoffier — APPROVED** [review]
Thanks

**Pull request merged**